### PR TITLE
Create backend behaviour

### PR DESCRIPTION
**Pull request by ShaneKilkelly**

### CODE REVIEW

File: doc_src/CreatingBackends.md [hunk 1]
```diff
@@ -3,6 +3,10 @@
 
 See `Hammer.Backend.ETS` for a realistic example of a Hammer Backend module.
 
+
+Backends for Hammer are expected to implement the [Hammer.Backend](/hammer/Hammer.Backend.html) behaviour.
```
Comment (sobolevn) [inline doc_src/CreatingBackends.md]: This link will be broken in many cases. It is better in my experience to have full urls.

Comment (ShaneKilkelly) [inline doc_src/CreatingBackends.md]: Acknowledged, I think I hacked this up quickly to get it working, will create a separate issue.

**Comment by sobolevn** [comment]
Nice, will start implementing new behavior soon.

**Comment by ShaneKilkelly** [comment]
I'll be updating the Redis module to follow the new behaviour too, hopefully later today.

**Pull request merged**